wireapp/wire-ios-mono
# fix: Sending logs when email client not configured - WPB-9948

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9948" title="WPB-9948" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9948</a>  Sending logs - when email client is not configured
  </td></table>
  <br />
 



















### Issue

Currently when sending logs via a share sheet we attempt to attach all logging URLs derived from LogFileDestination and ZMSLog. However in some cases no logs have actually been created, which unsurprisingly can cause sending logs to fail.

Originally I created a fix using the current implementation and filtering logging URLs which had no associated file. See https://github.com/wireapp/wire-ios/pull/1656. However I later realized that @David-Henner was working on a related ticket (https://github.com/wireapp/wire-ios/pull/1658) which gathers logs through a different method. I.e currently there are multiple ways of gathering debug logs. In this PR I fix the issue by using the same method of gathering logs as https://github.com/wireapp/wire-ios/pull/1658 uses.

This PR:

- Uses `LogFilesProvider.generateLogFilesZip` to gather logs instead of `DebugLogSender. debugLogs`. This has the side affect of fixing the bug.
- Deletes  `DebugLogSender.debugLogs` and related code.

### Testing

I'm not really sure of any reasonable way to write automated tests for this fix. If you have some ideas let me know. 

To manually test:

0. Ensure that Mail.app is not configured, but Gmail.app is.
1. Clean install and log in
2. Go to Account > Settings > Advanced > Debug Report > Send Report > Clic kOK on alert
3. Select Gmail and send logs
4. Check logs have been sent in Gmail.
5. Clean install and log in
6. Shake phone to go to Developer Tools
7. Select Debug Actions > Send Debug Report > Click OK on alert
8. Select Gmail and send logs
9. Check logs have been sent in Gmail.

### Checklist
